// __BEGIN_LICENSE__
//  Copyright (c) 2006-2013, United States Government as represented by the
//  Administrator of the National Aeronautics and Space Administration. All
//  rights reserved.
//
//  The NASA Vision Workbench is licensed under the Apache License,
//  Version 2.0 (the "License"); you may not use this file except in
//  compliance with the License. You may obtain a copy of the License at
//  http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// __END_LICENSE__


package vw.platefile;

message RpcNullMsg {
  optional int32 dummy = 1 [ default = 0 ];  // dummy parameter (empty message)
}

// The way enums work:
//   If you add an enum value to the list, code that doesn't known about that
//   new value will map it to the default. That default is UNKNOWN, which is
//   the same as "an exception that we don't specifically handle". You can add
//   to this enum, just don't re-use numbers.
message RpcErrorMsg {
  enum Code {
    // 0-100 reserved for system
    SUCCESS = 0;
    UNKNOWN = 1;
    REMOTE_ERROR = 2;

    // 101-200 reserved for plate
    PLATE_UNKNOWN = 101;
    PLATE_TILE_NOT_FOUND = 102;
    PLATE_INVALID = 103;
    PLATE_CREATION_FAILED = 104;
    PLATE_BLOB_IO = 105;
    PLATE_BLOB_LIMIT = 106;
  }

  optional Code code = 1 [ default = UNKNOWN ];
  optional string msg = 2;
}

message RpcWrapper {
  required string method     = 1;
  optional bytes payload     = 2;
  optional RpcErrorMsg error = 3;
  optional string requestor  = 4;
  optional int32 seq         = 5 [default = -1];
  required uint32 checksum   = 6;
}
